File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
def DFS(n):

    stack = [n]
    visited = [0] * (N + 1)

    while stack:
        m = stack.pop()
        if visited[m] == 0:
            visited[m] = 1
            dfs.append(m)
            for i in range(N, 0, -1):
                if arr[m][i] == 1 and visited[i] == 0:
                    stack.append(i)

def BFS(n):

    queue = [n]
    visited = [0] * (N + 1)

    while queue:
        m = queue.pop(0)
        if visited[m] == 0:
            visited[m] = 1
            bfs.append(m)
            for i in range(0, N+1):
                if arr[m][i] == 1 and visited[i] == 0:
                    queue.append(i)


N, M, V = map(int, input().split())

arr = [[0] * (N+1) for _ in range(N+1)]
dfs = []
bfs = []

for _ in range(M):
    a, b = map(int, input().split())
    arr[a][b] = 1
    arr[b][a] = 1

DFS(V)
print(*dfs)

BFS(V)
print(*bfs)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.js
let fs = require("fs");
let input = fs.readFileSync("/dev/stdin");
// let input = fs.readFileSync("./input.txt");
input = input.toString().trim().split("\n");

const [N, M, V] = input[0].split(" ").map((x) => parseInt(x));
let arr = Array.from({ length: N + 1 }, () => []);

for (let i = 1; i <= M; i++) {
  const [a, b] = input[i].split(" ").map((x) => parseInt(x));
  arr[a].push(b);
  arr[b].push(a);
}

function DFS(start) {
  let stack = [start];
  visited = new Array(N + 1).fill(0);
  ans = [];
  while (stack.length) {
    x = stack.pop();
    if (visited[x] === 0) {
      ans.push(x);
      visited[x] = 1;
      arr[x] = arr[x].sort((a, b) => b - a);
      for (let i = 0; i < arr[x].length; i++) {
        if (visited[arr[x][i]] === 0) {
          stack.push(arr[x][i]);
        }
      }
    }
  }
  return ans;
}

function BFS(start) {
  let queue = [start];
  visited = new Array(N + 1).fill(0);
  visited[start] = 1;
  ans = [];
  while (queue.length) {
    x = queue.shift();
    ans.push(x);
    arr[x] = arr[x].sort((a, b) => a - b);
    for (let i = 0; i < arr[x].length; i++) {
      if (visited[arr[x][i]] === 0) {
        visited[arr[x][i]] = 1;
        queue.push(arr[x][i]);
      }
    }
  }
  return ans;
}

const dfs = DFS(V);
const bfs = BFS(V);
console.log(...dfs);
console.log(...bfs);
